recirq/characterization/xeb/grid_parallel_two_qubit_xeb.py

```diff
@@ -0,0 +1,536 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Parallel two-qubit cross-entropy benchmarking on a grid.
+
+Cross-entropy benchmarking is a method of estimating the fidelity of quantum
+gates by executing random quantum circuits containing those gates.
+This experiment performs cross-entropy benchmarking of a two-qubit gate applied
+to connected pairs of grid qubits. The qubit pairs are benchmarked in parallel
+by executing circuits that act on many pairs simultaneously.
+"""
+
+from typing import (Any, Iterable, List, Optional, Sequence, TYPE_CHECKING,
+                    Tuple, cast)
+import collections
+from concurrent.futures import ThreadPoolExecutor
+import dataclasses
+import datetime
+import itertools
+import multiprocessing
+import os
+
+import numpy as np
+
+from cirq import devices, ops, protocols, sim, value
+from cirq.experiments.cross_entropy_benchmarking import (CrossEntropyResult,
+                                                         CrossEntropyResultDict,
+                                                         CrossEntropyPair,
+                                                         SpecklePurityPair)
+from cirq.experiments.fidelity_estimation import (
+    least_squares_xeb_fidelity_from_probabilities)
+from cirq.experiments.purity_estimation import purity_from_probabilities
+from cirq.experiments.random_quantum_circuit_generation import (
+    GridInteractionLayer,
```

Comment:
Doesn't this also belong to this experiment?

Reply:
It's always hard to draw the line, but I think there's appetite for keeping the library functionality in Cirq and having the "infrastructure" part that ties it all together in ReCirq.